Code Quality: Use compile time Regex

gumbarros opened the pull request:
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.

   Closes #15094, ~I'm aware if @d2dyno1 do not approve this, this PR can be closed~, he give the ok at Discord.

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

**Additional information**
All expressions are migrated _as is_ without any modification. I think this should have no impact in validations.
Open to suggestions for the namespace of `RegexHelpers`.

0x5bfa commented on src/Files.App/Data/Items/WindowsCompatibilityOptions.cs:
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License. See the LICENSE.
 
 using Files.Shared.Extensions;
+using RegexHelpers = Files.App.Data.Regex.RegexHelpers;
```
I think you can put it into GlobalUsings class.

gumbarros replied:
Will be better, because `Regex` namespace conflict with the `Regex` class, causing all this weird usings. Tomorrow I will fix

0x5bfa reviewed (commented):
Is this all?

gumbarros commented:
> Is this all?

Yes, every single Regex of the project

gumbarros commented:
*Every single compile time Regex, sorry for the screenshot, there just one exception

Line 100 of `BaseStorageItemQueryResult.cs`
```cs
	items = items.Where(x  => Regex.IsMatch(x.Name, colonSplit[1].Replace("\"", "", StringComparison.Ordinal).Replace("*", "(.*?)", StringComparison.Ordinal), RegexOptions.IgnoreCase)).ToList();
```
Line 105 of `BaseStorageItemQueryResult.cs`
```cs
							items = items.Where(x => Regex.IsMatch(x.Name, split.Replace("\"", "", StringComparison.Ordinal).Replace("*", "(.*?)", StringComparison.Ordinal), RegexOptions.IgnoreCase)).ToList();
```

They can be converted to:
```cs
// Line 100
items = FilterItems(items, colonSplit[1]);

// Line 105
items = FilterItems(items, split);

// Add this method to your class or a utility class
private List<StorageItem> FilterItems(List<StorageItem> items, string filter)
{
    var filteredItems = new List<StorageItem>();

    foreach (var item in items)
    {
        if (item.Name.IndexOf(filter, StringComparison.OrdinalIgnoreCase) >= 0)
        {
            filteredItems.Add(item);
        }
    }

    return filteredItems;
}
```
Right? But I would recommend a QA on this.

0x5bfa commented on src/Files.App/Utils/Storage/StorageItems/NativeStorageFile.cs:
```diff
@@ -261,5 +261,6 @@ public override IAsyncAction RenameAsync(string desiredName, NameCollisionOption
 
 		public override IAsyncOperation<StorageFile> ToStorageFileAsync()
 			=> throw new NotSupportedException();
+
```
```suggestion
```

0x5bfa commented on src/Files.App/Data/Items/WindowsCompatibilityOptions.cs:
```diff
@@ -1,7 +1,6 @@
 // Licensed under the MIT License. See the LICENSE.
 
-using Files.Shared.Extensions;
 
```
Remove doubled line break

0x5bfa reviewed (commented):
I think the requirements of the issue have already been met sufficiently.

0x5bfa reviewed (approved):
LGTM. Thank you! 🎉
